src/src/com/microsoft/aad/adal/DefaultTokenCacheStore.java

```diff
@@ -245,13 +245,15 @@ public void removeAll() {
      */
     @Override
     public HashSet<String> getUniqueUsersWithTokenCache() {
-        Iterator<TokenCacheItem> results = this.getAll();
+        Iterable<TokenCacheItem> results = this.getAll();
         HashSet<String> users = new HashSet<String>();
-
-        while (results.hasNext()) {
-            TokenCacheItem item = results.next();
-            if (item.getUserInfo() != null && !users.contains(item.getUserInfo().getUserId())) {
-                users.add(item.getUserInfo().getUserId());
+        
+        if (results != null) {
```

Comment:
Yes, it's safe when object can be null, and in this case it actually superficial code. (extra 2 lines + extra brackets + extra tabs for inner code)...